lmfdb/backend/statstable.py

```diff
@@ -753,6 +753,62 @@ def min(self, col, constraint={}, record=True):
                 self._record_extreme(col, ccols, cvals, m, kind="min")
         return m
 
+    def _slow_sum(self, col):
+        """
+        Compute the sum of a column
+
+        INPUT:
+
+        - ``col`` -- the column
+        """
+        inserter = SQL("SELECT SUM({0}) FROM {1}")
+        inserter = inserter.format(Identifier(col), Identifier(self.search_table))
+        cur = self._execute(inserter)
+        if cur.rowcount:
+            return cur.fetchone()[0]
+
+    def _record_sum(self, col, m):
+        """
+        Cache the sum
+
+        INPUT:
+
+        - ``col`` -- the column
+        - ``m``   -- the value to store as sum
+        """
+        inserter = SQL(
+                "INSERT INTO {0} "
+                "(cols, value) "
+                "VALUES (%s, %s)"
+            )
+        self._execute(
+            inserter.format(Identifier(self.stats)),
+            [Json([col]), m],
+        )
+
+    def _quick_sum(self, col):
+        """
+        This just looks up the sum; returns None if it hasn't been computed
+        """
+        values = [Json([col])]
+        selecter = SQL(
+            "SELECT value FROM {0} WHERE cols = %s AND threshold IS NULL"
+        ).format(Identifier(self.stats))
+        cur = self._execute(selecter, values)
+        if cur.rowcount:
+            return cur.fetchone()[0]        
+
+    def sum(self, col, record=True):
```

Comment:
As above, this should take `constraint` as an optional argument, like `max` does.

Reply:
done